### PR TITLE
Fix intermittent failure of test_e25519

### PR DESCRIPTION
**Pull request by seanyoung**
#### Problem
The test test_ed25519 fails occasionally. It's easy to reproduce this by running this in the sdk directory:
```
while cargo test test_ed25519; do : ; done
```
#### Summary of Changes
This PR fixes the test

Fixes #

### CODE REVIEW

**Comment by seanyoung** [comment]
I am wondering if we should also check if there are any trailing bytes in the instruction, or any unnecessary padding between the offsets/signatures/data.

**Comment by jstarry** [comment]
On further thought, I'm not sure this is a big deal. It will only be an issue if someone expects the full instruction data to be unique I think. But that seems unlikely. It seems much more likely that the signatures themselves would be used directly for something like that. That being said, if we still want to patch this up, it makes sense to prohibit trailing bytes too.

@CherryWorm any opinion here? Basically the issue is that we don't check the padding byte, so it can be anything... and then used as a nonce, replay attack against a really dumb implementation, etc.

**Comment by CherryWorm** [comment]
IIRC ed25519 signatures are not unique anyways, so even depending on the signature to be unique is not sufficient

File: sdk/src/ed25519_instruction.rs [hunk 1]
```diff
@@ -94,6 +98,9 @@ pub fn verify(
     let expected_data_size = num_signatures
         .saturating_mul(SIGNATURE_OFFSETS_SERIALIZED_SIZE)
         .saturating_add(SIGNATURE_OFFSETS_START);
+    if data[1] != 0 && feature_set.is_active(&check_ed25519_instruction_padding::id()) {
```
Comment (jackcmay) [inline sdk/src/ed25519_instruction.rs:101]: I see it is checked above but still iffy to use direct indexing, maybe use `get`

**Comment by jackcmay** [comment]
I don't see any harm in enforcing zero pre-pad and zero post-pad bytes unless we already have clients that expect othewise

**Comment by seanyoung** [comment]
There doesn't seem to be a compelling reason to check the padding for zero. So, just update the test.

**Review by jstarry — APPROVED** [review]
Looks good to me, I don't think it's a big deal that we ignore that byte.

**Pull request merged**